js/ui/diagram/ui.diagram.rightpanel.js

```diff
@@ -6,34 +6,56 @@ import DiagramCommands from "./ui.diagram.commands";
 import { extend } from "../../core/utils/extend";
 import DiagramBar from "./diagram_bar";
 
-const DIAGRAM_OPTIONS_CLASS = "dx-diagram-options";
-const OPTIONS = [
-    {
-        title: "Page Properties",
-        onTemplate: (widget, $element) => widget._renderOptions($element)
-    }
-];
+const DIAGRAM_RIGHT_PANEL_CLASS = "dx-diagram-right-panel";
 
-class DiagramOptions extends Widget {
+class DiagramRightPanel extends Widget {
     _init() {
         super._init();
         this.bar = new OptionsDiagramBar(this);
         this._valueConverters = {};
+        this._onDataToolboxRenderedAction = this._createActionByOption("onDataToolboxRendered");
     }
     _initMarkup() {
         super._initMarkup();
-        this.$element().addClass(DIAGRAM_OPTIONS_CLASS);
+        this.$element().addClass(DIAGRAM_RIGHT_PANEL_CLASS);
         const $accordion = $("<div>")
             .appendTo(this.$element());
 
         this._renderAccordion($accordion);
     }
+    _getDataSources() {
+        return this.option("dataSources") || {};
+    }
+    _getAccordionDataSource() {
+        var result = [{
+            title: "Page Properties",
+            onTemplate: (widget, $element) => widget._renderOptions($element)
+        }];
+        var dataSources = this._getDataSources();
+        var hasDataSources = false;
+        for(var key in dataSources) {
+            if(dataSources.hasOwnProperty(key)) {
```

Comment:
The `dataSources.hasOwnProperty(key)` construction always returns `true`, doesn't it?

Reply:
If somebody adds something to Ojbect via prototype this method will return false for it. 
Is there a better approach for hashes used in Devextreme?